config/browserstack.js

```diff
@@ -52,6 +52,47 @@ module.exports = function(grunt){
 	    })
 	    return results
 	}
+	
+	function isSameBrowser(browser1, browser2) {
+		if(JSON.stringify(browser1) === JSON.stringify(browser2)) {
+			return true;
+		}
+		return false;
+	}
+	
+	function hasBrowser(browserList, browser) {
+		return browserList.some(function(val){
+			return isSameBrowser(val, browser);
+		});
+	}
+	
+	function getConfigByBrowser() {
+		// moduleList getConfig
+		var configList = moduleList.map(function(val) {
+			return getConfig(val.name);
+		});
+		
+		var browsers = [];
+		for(var i in bsLaunchers) {
+			browsers = browsers.concat(bsLaunchers[i]);
+		}
```

Comment:
Could you use [`reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)?
```
var browsers = variable.reduce(function(a,b){
 return a.concat(b);
},[]);
```